### PR TITLE
feat: add avatar component + storybook example

### PR DESCRIPTION
**Pull request by atomictangerine**
## What I did

- Migrates the avatar component. 
- Adds storybook example
- Creates chromatic baseline
- Updates README 
- Adds avatar to the  Server Side Testing app page

Screen Shot 2022-06-13 at 9 17 03 PM
Displays avatar components on sandbox 
Screen Shot 2022-06-14 at 2 26 30 PM
When image is loading/not able to render
Screen Shot 2022-06-14 at 2 18 39 PM

### CODE REVIEW

File: ui/avatar/package.json [hunk 1]
```diff
@@ -0,0 +1,45 @@
+{
+	"name": "@washingtonpost/wpds-avatar",
+	"version": "0.9.1",
+	"description": "> TODO: description",
```
Comment (artmsilva) [inline ui/avatar/package.json:4]: Can we update this field to reflect the description from the design spec?

File: ui/avatar/src/Avatar.tsx [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import * as React from "react";
+import { styled } from "@stitches/react";
+import * as AvatarPrimitive from "@radix-ui/react-avatar";
+import { css, theme } from "@washingtonpost/wpds-theme";
```
Comment (artmsilva) [inline ui/avatar/src/Avatar.tsx:4]:
```suggestion
import { css, theme, styled } from "@washingtonpost/wpds-theme";
```

File: ui/avatar/src/play.stories.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
...
+      <img
+        src="https://i.pravatar.cc/300"
+        alt="An avatar is an atomic component that represents an individual’s identity through a circular photo."
+      />
+    </Component>
+    <Component {...args}>
+      <Image
+        loader={myLoader}
+        src="https://i.pravatar.cc/300"
+        width="32"
```
Comment (artmsilva) [inline ui/avatar/src/play.stories.jsx:30]: Can we document how to use next/image with Avatar in the ui/avatar/README.md for now?

File: ui/kit/package.json [hunk 4]
```diff
@@ -32,7 +28,9 @@
     "url": "https://github.com/washingtonpost/wpds-ui-kit/issues"
   },
   "dependencies": {
+    "@washingtonpost/wpds-avatar": "0.9.1",
     "@washingtonpost/eslint-plugin-wpds": "0.9.1",
+    "@washingtonpost/wpds-avatar": "0.9.1",
```
Comment (artmsilva) [inline ui/kit/package.json:33]: I think we can remove this since it's already declared above.

File: ui/kit/package.json [hunk 5]
```diff
@@ -58,7 +56,9 @@
     "typescript": "4.5.5"
   },
   "peerDependencies": {
+    "@washingtonpost/wpds-avatar": "*",
     "@washingtonpost/eslint-plugin-wpds": "*",
+    "@washingtonpost/wpds-avatar": "*",
```
Comment (artmsilva) [inline ui/kit/package.json:61]: Same here.

**Review by artmsilva — COMMENTED** [review]
Can we add examples to the server side testing page in the app? https://wpds-ui-kit-git-feature-create-avatar-component.preview.now.washingtonpost.com/. This helps us catch possible server side render issues with our components.

**Comment by artmsilva** [comment]
Also your PR title and commit messages should be updated to say "feat: add avatar component + storybook example". See here for spec https://www.conventionalcommits.org/en/v1.0.0/#summary

File: ui/avatar/src/play.stories.jsx (hunk 3 above)
Comment (atomictangerine) [reply ui/avatar/src/play.stories.jsx:30]: Good idea :) I was going to write it in the docs (and thinking about making them next and all that) so the README slipped my mind.

File: ui/avatar/src/Avatar.tsx [hunk 6]
```diff
@@ -0,0 +1,63 @@
...
+const NAME = "Avatar";
+const DEFAULT_AVATAR_SIZE = "200";
+
+interface AvatarProps {
+  /**
+   * Sizes - supports any size token
+   * @default 200
+   *
+   * */
+  size?: string;
```
[the remark this answers is not in the document]
Comment (atomictangerine) [reply ui/avatar/src/Avatar.tsx]: updated to follow the Icon component

File: app/pages/index.tsx [hunk 7]
```diff
@@ -317,6 +318,13 @@ function HomePage() {
         </HStack>
         <h2>Divider</h2>
         <Divider />
+        <h2>Avatar</h2>
+        <Avatar size="200">
+          <img
+            src="https://i.pravatar.cc/300"
+            alt="An avatar is an atomic component that represents an individual’s identity through a circular photo."
+          />
+        </Avatar>
```
Comment (artmsilva) [inline app/pages/index.tsx:327]: Can we add the next.js image example here too?

File: ui/avatar/package.json [hunk 8]
```diff
@@ -0,0 +1,45 @@
...
+	},
+	"repository": {
+		"type": "git",
+		"url": "git+https://github.com/washingtonpost/wpds-ui-kit.git"
+	},
+	"scripts": {
+		"test": "echo \"Error: run tests from root\" && exit 1",
+		"build": "tsup src/index.ts --loader .ts=tsx --minify --format esm,cjs --dts --sourcemap --legacy-output --external react",
+		"dev": "tsup src/index.ts --format esm,cjs --watch --dts --legacy-output --external react",
+    	"clean": "rm -rf .turbo && rm -rf node_modules && rm -rf dist"
```
Comment (artmsilva) [inline ui/avatar/package.json:29]: Can we run prettier on these files?

File: app/next.config.js [hunk 9]
```diff
@@ -10,4 +11,7 @@ module.exports = withBundleAnalyzer({
     },
   ],
   swcMinify: false,
+  images: {
+    domains: ["i.pravatar.cc/300"],
```
Comment (artmsilva) [inline app/next.config.js:15]:
```suggestion
    domains: ["i.pravatar.cc"],
```

File: ui/avatar/src/play.stories.jsx [hunk 10]
```diff
@@ -0,0 +1,51 @@
...
+
+const myLoader = ({ src }) => {
+  return `${src}`;
+};
+
+const DefaultArgs = {
+  size: "200",
+};
+
+const Container = styled("h3", {
```
Comment (artmsilva) [inline ui/avatar/src/play.stories.jsx:19]: Should this be a section or div? I assume this was a copypasta mistake

File: ui/avatar/src/Avatar.tsx [hunk 11]
```diff
@@ -0,0 +1,62 @@
...
+    <StyledAvatar
+      ref={ref}
+      css={{
+        width: theme.sizes[size],
+        height: theme.sizes[size],
+      }}
+    >
+      {React.cloneElement(child, {
+        className: `${ImageStyles}`,
+        ...props,
```
Comment (wp-aberg) [inline ui/avatar/src/Avatar.tsx:56]: Having the Avatar's props assigned to the image might yield unexpected results. Especially if the same prop is passed and the image's prop gets overwritten.

**Review by wp-aberg — COMMENTED** [review]
Overall I think this looks good, just the one comments about where the props end up

File: ui/avatar/src/Avatar.tsx [hunk 12]
```diff
@@ -0,0 +1,61 @@
...
+const StyledAvatar = styled(AvatarPrimitive.Root, {
+  display: "inline-flex",
+  alignItems: "center",
+  justifyContent: "center",
+  verticalAlign: "middle",
+  overflow: "hidden",
+  userSelect: "none",
+  borderRadius: "100%",
+  backgroundColor: theme.colors.subtle,
+  marginRight: theme.sizes["050"],
```
Comment (artmsilva) [inline ui/avatar/src/Avatar.tsx:27]: I think we want to leave layout decisions to the user. Can we remove the marginRight and marginBottom?

File: ui/avatar/src/Avatar.tsx [hunk 13]
```diff
@@ -0,0 +1,61 @@
...
+    borderRadius: "inherit",
+    length: 0,
+  });
+
+  return (
+    <StyledAvatar
+      ref={ref}
+      css={{
+        width: theme.sizes[size],
+        height: theme.sizes[size],
```
Comment (artmsilva) [inline ui/avatar/src/Avatar.tsx:51]: We want to give users the ability to override or change our CSS through a `css` prop. See other components for examples.

Comment (artmsilva) [reply ui/avatar/src/Avatar.tsx:51]: https://build.washingtonpost.com/resources/guides/contribute-a-component#Core%20Tenets 

> We should should allow modification of styles via the css prop using our tokens (component (locally scoped and global) and theme).

File: ui/avatar/src/Avatar.tsx (hunk 12 above)
Comment (atomictangerine) [reply ui/avatar/src/Avatar.tsx:27]: I was following the design spec about the minimum spacing. But yes, we can remove that.

Screen Shot 2022-06-17 at 1 22 46 PM

Comment (atomictangerine) [reply ui/avatar/src/Avatar.tsx:27]: The zeplin doc is very confusing on what we want in code and what's for users/designers

Comment (artmsilva) [reply ui/avatar/src/Avatar.tsx:27]: @atomictangerine  I can see that. The "Recommended spacing" is under the section heading "Guidance". I think we can work with @unconstruct to refine the heading (maybe add a description) to reflect the intent which is for feature teams and how they use our components (Do's and Don'ts)

**Review by clairehhelms — APPROVED** [review]

**Comment by artmsilva** [comment]
oof looks like we got conflicts and some checks failing

File: ui/avatar/src/Avatar.tsx [hunk 14]
```diff
@@ -55,6 +55,7 @@ export const Avatar: React.FC<AvatarProps> = React.forwardRef<
     >
       {React.cloneElement(child, {
         className: `${ImageStyles}`,
+        tabIndex: 0,
```
Comment (artmsilva) [inline ui/avatar/src/Avatar.tsx:58]: why is this keyboard focusable? curious

**Pull request merged**